plugins/package-managers/node/src/main/kotlin/Yarn.kt

```diff
@@ -88,9 +87,9 @@ class Yarn(
 
         val process = run(workingDir, "info", "--json", packageName)
 
-        val data = Json.parseToJsonElement(process.stdout).jsonObject["data"]?.also {
+        val data = parseJsonLines(process.stdout).jsonObject["data"]?.also {
```

Comment:
Can't this be solved by just using `Json.decodeToSequence().last()`?

Reply:
The `decodeToSequence()` function expects that all objects are of the same type. This is not the case here; the objects representing retries have different properties. Also, I am not sure whether Yarn always uses this single-object-per-line format. I wanted to stay as compatible with the change as possible. Therefore, the different way of parsing is only tried as a fallback after the normal parsing has failed.

Reply:
> The `decodeToSequence()` function expects that all objects are of the same type.

Well, yes, but that type could be `JsonElement` like at

https://github.com/oss-review-toolkit/ort/blob/bb336f0e57f273ea9880e4566d33b8657768eb21/plugins/package-managers/node/src/main/kotlin/PackageJson.kt#L49-L52

> Also, I am not sure whether Yarn always uses this single-object-per-line format.

I believe it's safe to assume so until we come across a counter-example. The current heuristic also makes some assumptions, so if in question, I'd prefer to go with simpler code that makes use of `decodeToSequence()`.